.github/workflows/release.yml

```diff
@@ -1,101 +1,61 @@
+  release-please:
+    name: release
     runs-on: ubuntu-latest
+
     steps:
       - name: Ensure main branch
         if: github.ref != 'refs/heads/main'
         run: |-
           echo "Not running on main - exit"
           exit 1
 
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         with:
           fetch-depth: 0
 
-      - name: 🔍 Check if version changed
-        uses: EndBug/version-check@v1
-        if: github.event_name == 'push'
-        id: check
-
-      - name: 🔄 Check if should release
-        run: echo "SHOULD_RELEASE=${{ steps.check.outputs.changed == 'true' || github.event_name == 'workflow_dispatch' }}" >> $GITHUB_ENV
-
       - name: ⚙️ Setup Node.js v16
         uses: actions/setup-node@v2
-        if: env.SHOULD_RELEASE == 'true'
         with:
           node-version: '16.x'
           registry-url: 'https://registry.npmjs.org'
           scope: '@erc725'
           cache: 'npm'
 
-      - name: 📝 Set Version
-        if: env.SHOULD_RELEASE == 'true'
-        run: |-
-          APP_VERSION="v$(node -pe "require('./package.json').version")"
-          echo "APP_VERSION=$APP_VERSION" >> $GITHUB_ENV
-
       - name: 🧰 Install
-        if: env.SHOULD_RELEASE == 'true'
         run: npm ci
 
+      - name: 💅 Lint
+        run: npm run lint
+
       - name: 🎯 Test
-        if: env.SHOULD_RELEASE == 'true'
         run: npm test
 
       - name: 🛠 Build
-        if: env.SHOULD_RELEASE == 'true'
         run: npm run build
 
-      # We assume this will be always triggered by a merge from develop
-      # Therefore we tag the previous commit (the merge commit won't be on develop)
-      - name: 🏷 Create and push Git Tag
-        if: env.SHOULD_RELEASE == 'true'
-        run: |-
-          git config --global user.email "release@lukso.network"
-          git config --global user.name "LUKSO Bot"
-          git tag -a ${{ env.APP_VERSION }} HEAD~ -m "Release Version ${{ env.APP_VERSION }} [CI]"
-          git push --set-upstream origin tag ${{ env.APP_VERSION }}
-
-      # Create GitHub Release
-      - name: 📝 Extract release notes from CHANGELOG
-        if: env.SHOULD_RELEASE == 'true'
-        id: extract-release-notes
-        uses: ffurrer2/extract-release-notes@v1
-        with:
-          release_notes_file: RELEASENOTES.md
-
-      - uses: jwalton/gh-find-current-pr@v1
-        if: env.SHOULD_RELEASE == 'true'
-        id: findPR
+      - uses: google-github-actions/release-please-action@v3
+        id: release
         with:
-          state: closed
+          release-type: node
+          package-name: '@erc725/erc725.js'
+          bump-minor-pre-major: true
+          default-branch: main
 
-      - name: Add PR body to Release Notes
-        if: env.SHOULD_RELEASE == 'true'
-        env:
-          PR_BODY: ${{ steps.findPR.outputs.body }}
-        run: |-
-          echo ${PR_BODY}|cat - RELEASENOTES.md > /tmp/out && mv /tmp/out RELEASENOTES.md
+      - name: 🔍 Check if version changed
```

Comment:
i think this could go in a separate workflow.

One for release please
One for publish to npm

but lets keep it like this for now